Sources/TuistGenerator/Linter/GraphLinter.swift

```diff
@@ -442,6 +443,7 @@ public class GraphLinter: GraphLinting {
             LintableTarget(platform: .macOS, product: .dynamicLibrary),
             LintableTarget(platform: .macOS, product: .staticFramework),
             LintableTarget(platform: .macOS, product: .framework),
+            LintableTarget(platform: .macOS, product: .app),
```

Comment:
It's a matter of dependency graph. 
CleanMyMac consists of 3 apps (actually, a bit more):
- Main app
- Menu app
- HealthMonitor app
<img width="425" alt="image" src="https://user-images.githubusercontent.com/2575555/212397193-472faa9d-bf27-4425-8ae6-4ca2e7346a48.png">

Our old dependency graph looked like this:
```mermaid
graph TD;
    Main-->Menu;
    Main-->HealthMonitor;
```
Main app waits for other apps to be built.

Our new dependency graph looks like this:
```mermaid
graph TD;
    Aggregate-->Main;
    Aggregate-->Menu;
    Aggregate-->HealthMonitor;
```
All apps are compiled in parallel. Once everything is built, each app is put in the proper directory. 
I'm not sure if we can archive this behaviour using schemes.

Reply:
Thanks for clarifying @serejahh 

Based on your explanation, it seems you're after creating an aggregate target rather than a command line tool that actually has a dependency on those applications.

![aggregate-target](https://user-images.githubusercontent.com/11914919/212848776-98755282-b88d-4c3c-9abe-5737eb3b3fc7.png)

Xcode supports this, but Tuist does not yet have support for this (see https://github.com/tuist/tuist/issues/3388) 😞

I believe schemes may be able to help if the goal is to build them all in parallel. I did a local test

![Screenshot 2023-01-17 at 08 33 46](https://user-images.githubusercontent.com/11914919/212849938-dfb38885-0883-4d2b-88dd-7a823ead8823.png) (in this example I only added the applications to the scheme but not as target dependencies)

Inspecting the build timeline we can those targets are built in parallel as they don't depend on each other

![scheme timeline](https://user-images.githubusercontent.com/11914919/212850172-9ee25992-7825-4e55-b574-0af73774165c.png)

There's a scheme example which may be of use 
https://github.com/tuist/tuist/blob/main/projects/tuist/fixtures/ios_app_with_custom_scheme/Workspace.swift

Reply:
I agree that if it's just for building a scheme should be enough

Reply:
Thanks, I'll investigate if it works for us

Reply:
What do you think if I change the PR for `macOS` -> `macOS` linting? Does it look to be merged?

Reply:
@serejahh can you clarify which specific products in `macOS` ?

Reply:
I've updated the issue and PR by removing CLI and mac apps inside. Please check